pkg/controller/generic_reconciler.go

```diff
@@ -177,7 +191,6 @@ func (gr *GenericReconciler) reconcileEverything(ctx context.Context) error {
 			"Kind", resource.Kind)
 	}
 
-	gr.watchNamespaces.resetCache()
 	namespaces, err := gr.watchNamespaces.getWatchNamespaces(ctx, gr.client)
```

Comment:
Do we need to do this twice then? Perhaps we could send the list of namespaces to the `reconcileEverything`?

Reply:
`getWatchNamespaces` function checks if the namespaces are set, if they are it only returns the namespaces collection. That's why I changed the cache reset to the start of the loop.

→ https://github.com/ncaak/deployment-validation-operator/blob/DVO-209/cpu-leak-extract-getWatchNamespaces/pkg/controller/watchnamespaces.go#L75

Reply:
Ohh OK...it's a bit confusing